File: leetcode/cpp_unittest/longestPalindrome_unittest/longestPalindrome_unittest.cpp
#include "longestPalindrome.cpp"
#include "gtest/gtest.h"

/*
 * string Solution::maxpalindrome(const string&)
 * string Solution::Manache(const string& S)
 * */

TEST(maxpalindromeTest, Positive01){
    Solution s;

    const string str("dbdogodcde");;
    string expected("dogod");

    string result = s.maxpalindrome(str);
    EXPECT_STREQ(expected.c_str(), result.c_str());
}

TEST(maxpalindromeTest, Positive02){
    Solution s;

    const string str("dbdogodcggddggsgsdghhsdfsdfe");;
    string expected("ggddgg");

    string result = s.maxpalindrome(str);
    EXPECT_STREQ(expected.c_str(), result.c_str());
}

TEST(ManacherTest, Positive01){
    Solution s;

    const string str("dbdogodcde");;
    string expected("dogod");

    string result = s.Manacher(str);
    EXPECT_STREQ(expected.c_str(), result.c_str());
}

TEST(ManacherTest, Positive02){
    Solution s;

    const string str("dbdogodcggddggsgsdghhsdfsdfe");;
    string expected("ggddgg");

    string result = s.Manacher(str);
    EXPECT_STREQ(expected.c_str(), result.c_str());
}

File: leetcode/java_unittest/longestPalindrome_junit/longestPalindromeUnitTest.java
/*
 * String longestPalindrome.Manacher(String str)
 *
 * */

import java.util.ArrayList;
import java.util.Arrays;
import org.junit.*;
import org.junit.Assert;

public class longestPalindromeUnitTest{
    private longestPalindrome game;

    @Before
    public void setUp(){
        game = new longestPalindrome();
    }

    @Test
    public void Manacher_positive01(){
        final String str = new String("dbdogodcde");
        final String expected = new String("dogod");
        String result = game.Manacher(str);

        Assert.assertEquals(expected, result);
    }

    @Test
    public void Manacher_positive02(){
        final String str = new String("dbdogodcggddggsgsdghhsdfsdfe");;
        final String expected = new String("ggddgg");
        String result = game.Manacher(str);

        Assert.assertEquals(expected, result);
    }

    @After
    public void settleDown(){
        game = null;
    }
}
